src/mongocxx/options/index.cpp

```diff
@@ -0,0 +1,172 @@
+    _v = v;
+}
+
+void index::weights(bsoncxx::document::view weights) {
+    _weights = weights;
+}
+
+void index::default_language(std::string default_language) {
+    _default_language = default_language;
+}
+
+void index::language_override(std::string language_override) {
+    _language_override = language_override;
+}
+
+void index::partial_filter_expression(bsoncxx::document::view partial_filter_expression) {
+    _partial_filter_expression = partial_filter_expression;
+}
+
+void index::twod_sphere_version(std::uint8_t twod_sphere_version) {
+    _twod_sphere_version = twod_sphere_version;
+}
+
+void index::twod_bits_precision(std::uint8_t twod_bits_precision) {
+    _twod_bits_precision = twod_bits_precision;
+}
+
+void index::twod_location_min(double twod_location_min) {
+    _twod_location_min = twod_location_min;
+}
+
+void index::twod_location_max(double twod_location_max) {
+    _twod_location_max = twod_location_max;
+}
+
+void index::haystack_bucket_size(double haystack_bucket_size) {
+    _haystack_bucket_size = haystack_bucket_size;
+}
+
+const stdx::optional<bool>& index::background() const {
+    return _background;
+}
+
+const stdx::optional<bool>& index::unique() const {
+    return _unique;
+}
+
+const stdx::optional<std::string>& index::name() const {
+    return _name;
+}
+
+const stdx::optional<bool>& index::sparse() const {
+    return _sparse;
+}
+
+const index::base_storage_options* index::storage_options() const {
+    return _storage_options;
+}
+
+const stdx::optional<std::int32_t>& index::expire_after_seconds() const {
+    return _expire_after_seconds;
+}
+
+const stdx::optional<std::int32_t>& index::v() const {
+    return _v;
+}
+
+const stdx::optional<bsoncxx::document::view>& index::weights() const {
+    return _weights;
+}
+
+const stdx::optional<std::string>& index::default_language() const {
+    return _default_language;
+}
+
+const stdx::optional<std::string>& index::language_override() const {
+    return _language_override;
+}
+
+const stdx::optional<bsoncxx::document::view>& index::partial_filter_expression() const {
+    return _partial_filter_expression;
+}
+
+const stdx::optional<std::uint8_t>& index::twod_sphere_version() const {
+    return _twod_sphere_version;
+}
+
+const stdx::optional<std::uint8_t>& index::twod_bits_precision() const {
+    return _twod_bits_precision;
+}
+
+const stdx::optional<double>& index::twod_location_min() const {
+    return _twod_location_min;
+}
+
+const stdx::optional<double>& index::twod_location_max() const {
+    return _twod_location_max;
+}
+
+const stdx::optional<double>& index::haystack_bucket_size() const {
+    return _haystack_bucket_size;
+}
+
+void index::wiredtiger_storage_options::config_str(std::string config_str) {
+    _config_str = config_str;
+}
+
+const stdx::optional<std::string>& index::wiredtiger_storage_options::config_str() const {
+    return _config_str;
+}
+
+index::wiredtiger_storage_options::~wiredtiger_storage_options() = default;
```

Comment:
Try to keep definitions in .cpp files in the same order as declarations in the header